types/react/test/index.ts

```diff
@@ -440,6 +440,17 @@ function RefCarryingComponent() {
     );
 }
 
+const MemoizedForwardingRefComponent = React.memo(ForwardingRefComponent);
+const LazyRefComponent = React.lazy(() => Promise.resolve({ default: RefComponent }));
+
+type RefComponentAsRef = React.ElementRef<typeof RefComponent>; // $ExpectType RefComponent
+type RefCarryingComponentAsRef = React.ElementRef<typeof RefCarryingComponent>; // $ExpectType undefined
```

Comment:
What does this return for a function component that expects props?

Reply:
Good catch! This indeed didn’t work after the change to `() => ReactElement | null`.